src/node/mod.rs

```diff
@@ -394,6 +402,15 @@ impl Node {
         Ok(())
     }
 
+    /// Returns the current BLS secret key share or `RoutingError::InvalidState` if we are not
+    /// elder.
+    pub fn secret_key_share(&self) -> Result<&bls::SecretKeyShare> {
+        self.stage
+            .approved()
+            .and_then(|stage| stage.secret_key_share())
```

Comment:
is that only an elder will have a valid secet_key_share ?
if so, maybe make it more explicit in the comment?

Reply:
Either elders or nodes that are in the process of being promoted to elders and have already seen the DKG result but have not yet seen the `SectionInfo` event. WIll add some comment there.

Reply:
it's just the function name of `approved` misleading me to think that an `adult` may have a `secret_key_share` as well.
yeah, give it a second thought and I don't get a better suggestion on comment here.
so, just leave it then.

Reply:
Yeah, it's been brought up before that `approved` might not be the best name, because there is no real approval going on. Should probably be renamed to `joined` or something like that.